### ISSUE TITLE
Daylight savings time does not seem to be handled correctly.

### ISSUE THREAD

**Issue by tchari** [fightbulc/moment.php#146]
When creating a new moment at hour when daylight savings changes, Moment returns an invalid date error. Moment.js, on the other hand, returns the changed hour. For example:
`$date = new Moment('2017-03-12 02:00:00', 'America/Edmonton);`
return an invalid date. However, in moment.js
`var now = moment('2017-03-12 02:00:00').format('MMM DD h:mm A');`
returns "Mar 12 3:00 AM"

I believe the isValid() is unable to handle the daylight savings hour case.

**Comment by hriziya** [comment]
I wanted to use this library for my ongoing project but I am unsure on how this will handle the DST?

**Comment by tchari** [comment]
For my use case I was taking as input a date time string (e.g. `2018-04-05 15:00:00`) and a time zone separately. I had to create a native `DateTime` and `DateTimeZone` object first and then create my `Moment` from the `DateTime`.

**Comment by hriziya** [comment]
I agree, but I am still not sure if creating the native `DateTime`  and `DateTimeZone` would work.. 

here is the small sample I am worried about. 
I am trying to convert `America/Chicago` time to `UTC` to store in database. but you can see that when I try to consider `Timezone Name` and `Timezone Offset` the output are different event with native `DateTimeZone` object. so I doubt Moment could handle this. This may not be the DST issue. but something related to it.
![sample](https://user-images.githubusercontent.com/4135216/53812223-0a70b380-3f81-11e9-9ac8-f6b549777402.png)
Any clue on this?

**Comment by tchari** [comment]
@hriziya 
Super late on this reply; sorry

I don't think using the offset strategy vs. a named timezone is really comparable because there are a lot of different timezones that the same offset.

For example, lets say you were using a an offset of -7. In the summer months that offset corresponds to America/Whitehorse, America/Vancouver, America/Los_Angeles, among others. In the winter months, that offset corresponds to America/Whitehorse, America/Edmonton, America/Denver, etc.

If you're using a fixed offset would not expect DLS to ever really be handled appropriately because DLS is a jurisdiction-by-jurisdiction thing. I think if you use the named timezone, native PHP's DateTime and DateTimeZone classes should handle it properly.
